### PR TITLE
feat: add vercel example

### PR DESCRIPTION
**Pull request by yusinto**
This adds a nextjs edge config example app.

yarn workspaces don't seem to work for nextjs so I still have to work on linking local packages to the example-vercel app. In the meantime I had to resort to manually copying over dist files to example-vercel which is very bad.

But this does show that the vercel code in #108 works without the need for `esModuleInterop`.

### CODE REVIEW

File: packages/sdk/vercel/example-vercel/src/app/api/hello/route.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export async function GET(request: NextRequest) {
+  const sdkKey = 'test-sdk-key';
+  const flagKey = 'testFlag1';
+
+  // default fallthrough evaluates to true
+  const contextFallthrough = { kind: 'user', key: 'test-user-key-1' };
+
+  // matches email targeting rule evaluates to false
+  const contextEmailRule = { kind: 'user', key: 'test-user-key-1', email: 'test@gmail.com' };
```
Comment (yusinto) [inline packages/sdk/vercel/example-vercel/src/app/api/hello/route.ts:17]: This is using the data from utils/testData.json and it looks like this in my edge config:

```json
{
  "LD-Env-test-sdk-key": {
    "flags": {
      "testFlag1": {
        "key": "testFlag1",
        "on": true,
        "prerequisites": [],
        "targets": [],
        "rules": [
          {
            "variation": 1,
            "id": "rule1",
            "clauses": [
              {
                "contextKind": "user",
                "attribute": "/email",
                "op": "contains",
                "values": [
                  "gmail"
                ],
                "negate": false
              }
            ],
            "trackEvents": false,
            "rollout": {
              "bucketBy": "bucket",
              "variations": [
                {
                  "variation": 1,
                  "weight": 100
                }
              ]
            }
          }
        ],
        "fallthrough": {
          "variation": 0
        },
        "offVariation": 1,
        "variations": [
          true,
          false
        ],
        "clientSideAvailability": {
          "usingMobileKey": true,
          "usingEnvironmentId": true
        },
        "clientSide": true,
        "salt": "aef830243d6640d0a973be89988e008d",
        "trackEvents": false,
        "trackEventsFallthrough": false,
        "debugEventsUntilDate": null,
        "version": 2,
        "deleted": false
      },
      "testFlag2": {
        "key": "testFlag2",
        "on": true,
        "prerequisites": [],
        "targets": [],
        "rules": [],
        "fallthrough": {
          "variation": 0,
          "rollout": {
            "bucketBy": "bucket",
            "variations": [
              {
                "variation": 1,
                "weight": 100
              }
            ],
            "contextKind:": "user",
            "attribute": "/email"
          }
        },
        "offVariation": 1,
        "variations": [
          true,
          false
        ],
        "clientSideAvailability": {
          "usingMobileKey": true,
          "usingEnvironmentId": true
        },
        "clientSide": true,
        "salt": "aef830243d6640d0a973be89988e008d",
        "trackEvents": false,
        "trackEventsFallthrough": false,
        "debugEventsUntilDate": null,
        "version": 2,
        "deleted": false
      },
      "testFlag3": {
        "key": "testFlag3",
        "on": true,
        "prerequisites": [],
        "targets": [],
        "rules": [
          {
            "variation": 1,
            "id": "rule1",
            "clauses": [
              {
                "op": "segmentMatch",
                "values": [
                  "testSegment1"
                ],
                "negate": false
              }
            ],
            "trackEvents": false
          }
        ],
        "fallthrough": {
          "variation": 0
        },
        "offVariation": 1,
        "variations": [
          true,
          false
        ],
        "clientSideAvailability": {
          "usingMobileKey": true,
          "usingEnvironmentId": true
        },
        "clientSide": true,
        "salt": "aef830243d6640d0a973be89988e008d",
        "trackEvents": false,
        "trackEventsFallthrough": false,
        "debugEventsUntilDate": null,
        "version": 2,
        "deleted": false
      }
    },
    "segments": {
      "testSegment1": {
        "name": "testSegment1",
        "tags": [],
        "creationDate": 1676063792158,
        "key": "testSegment1",
        "included": [],
        "excluded": [],
        "includedContexts": [],
        "excludedContexts": [],
        "_links": {
          "parent": {
            "href": "/api/v2/segments/default/test",
            "type": "application/json"
          },
          "self": {
            "href": "/api/v2/segments/default/test/beta-users-1",
            "type": "application/json"
          },
          "site": {
            "href": "/default/test/segments/beta-users-1",
            "type": "text/html"
          }
        },
        "rules": [
          {
            "id": "rule-country",
            "clauses": [
              {
                "attribute": "country",
                "op": "in",
                "values": [
                  "australia"
                ],
                "negate": false
              }
            ]
          }
        ],
        "version": 1,
        "deleted": false,
        "_access": {
          "denied": [],
          "allowed": []
        },
        "generation": 1
      },
      "testSegment2": {
        "name": "testSegment2",
        "tags": [],
        "creationDate": 1676063792158,
        "key": "testSegment2",
        "included": [],
        "excluded": [],
        "includedContexts": [],
        "excludedContexts": [],
        "_links": {
          "parent": {
            "href": "/api/v2/segments/default/test",
            "type": "application/json"
          },
          "self": {
            "href": "/api/v2/segments/default/test/beta-users-1",
            "type": "application/json"
          },
          "site": {
            "href": "/default/test/segments/beta-users-1",
            "type": "text/html"
          }
        },
        "rules": [],
        "version": 1,
        "deleted": false,
        "_access": {
          "denied": [],
          "allowed": []
        },
        "generation": 1
      }
    }
  }
}
```

File: packages/sdk/vercel/example-vercel/src/app/api/hello/route.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  const vercelClient = createClient(process.env.EDGE_CONFIG);
```
Comment (yusinto) [inline packages/sdk/vercel/example-vercel/src/app/api/hello/route.ts:19]: Make sure you set your `.env.development.local` EDGE_CONFIG to your own endpoint.

**Review by ldhenry — APPROVED** [review]
I didn't try it out personally but it looks good to me!

**Comment by yusinto** [comment]
Closing this in favor of #117

**Pull request closed without merging**